### PR TITLE
test: add integration tests

### LINKED ISSUE #2: Crate should not default select kubernetes versions and kube features
Currently `Cargo.toml` sets default features enabling many kube features and also k8s-openapi features:

https://github.com/kube-rs/gateway-api-rs/blob/main/Cargo.toml#L13-L20

We should try to minimize these as much as possible to:

1. avoid making dependency problems worse for users if we lag behind updates
2. avoid making the k8s-openapi version feature selection any more confusing than it is

For 2. we shoul definitely not set the k8s-openapi version feature by default, because then if users pick another version, they will get the compile error about multiple version features.

For 1. you could limit almost all kube features to be dev-dependencies so that we can test them, but avoid forcing them on users by default.

You do probably need to actually depend on kube with the `derive` feature enabled though. The don't see a smart way around that, except possibly generating the code with `--hide-kube`, but then users need to newtype the main structs and do the `#[derive(CustomResource)]` themselves on these (which possibly changes the schemas because it's wrapped, and makes it harder to test things here).

### PR DESCRIPTION
**Pull request by shaneutt**
This PR adds some initial integration tests and adds them to ci workflows, and also includes some `Cargo.toml` cleanup according to #2, as the initial POC had an overloaded copy/pasted `Cargo.toml`.

Resolves #2

### CODE REVIEW

**Comment by shaneutt** [comment]
@clux when you have some time if you wouldn't mind taking a look at this one and make sure I've got everything you had in mind for #2 done, I would appreciate it :vulcan_salute:

File: src/lib.rs [hunk 1]
```diff
@@ -1 +1,182 @@
...
+        name: String,
+    }
+
+    impl Drop for Cluster {
+        fn drop(&mut self) {
+            match delete_kind_cluster(&self.name) {
+                Err(err) => panic!("failed to cleanup kind cluster {}: {}", self.name, err),
+                Ok(()) => {}
+            }
+        }
```
Comment (clux) [inline src/lib.rs:90]: hah, this is cool

**Review by clux — APPROVED** [review]

**Comment by clux** [comment]
Looks sensible to me!

File: src/lib.rs [hunk 2]
```diff
@@ -1 +1,182 @@
 pub mod apis;
+
+#[cfg(test)]
+#[cfg(feature = "run-integration-tests")]
```
[the remark this answers is not in the document]
Comment (shaneutt) [reply src/lib.rs:4]: Sounds good, I'll make that change to be in line with the rest of kube-rs :+1:

[the remark this answers is not in the document]
Comment (shaneutt) [reply src/lib.rs]: switched up in 25fb749f1c37a1b5cf1e58448acb08c901fa9f56 :+1:

**Review by clux — APPROVED** [review]

**Pull request merged**